Initium-Odp/src/com/universeprojects/miniup/server/controllers/HCMLeaderboardController.java

```diff
@@ -51,6 +51,8 @@ protected String processRequest(HttpServletRequest request, HttpServletResponse
 		for(int i = 0; i<characters.size(); i++)
 		{
 			CachedEntity character = characters.get(i);
+						
+			if((int)character.getProperty("hardcoreRank")<=0||character.getProperty("hardcoreRank")==null) continue;
```

Comment:
it does the checks in left to right order, so it would never check the null before crashing on the (int) cast.

Also, you should add some spaces in there specifically around the || and optionally around the <= and ==